solutions/ac-SmallMultiple.mdx

```diff
@@ -0,0 +1,61 @@
+---
+id: ac-SmallMultiple
+source: AC
+title: Small Multiple
+author: Chongtian Ma
+---
+
+[Official Analysis](https://img.atcoder.jp/arc084/editorial.pdf)
+
+## Implementation
+
+<LanguageSection>
+
+<CPPSection>
+  
+**Time Complexity:** $\mathcal{O}(K)$
+  
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+const int INF = 1e9;
```

Comment:
why not INT32_MAX

Reply:
why INT32_MAX?

Reply:
well, it's the max value of an int?

Reply:
the max value of an int isnt necessary
just something big enough is fine

Reply:
> i dont think anyone would use INT32_MAX out of their own will lol

??? elaborate?

Reply:
im saying it's not good practice to use INT32_MAX (because sometimes overflow issues and its a lot to type out lol)

also its not necessary so why are you making me use this?

Reply:
> im saying it's not good practice to use INT32_MAX

give link which says so

> because sometimes overflow issues

why not check whether a value if MAX before adding? it's not like your 1 billion value can't overflow

> its a lot to type out lol

...autocomplete...

Reply:
> you've used it in your other PRs, why are you so against it now?

INF as a constant is just simply better (shorter and clearer). What more do you want me to say?

Reply:
> you've used it in your other PRs

so why the change of mind?

Reply:
> 

i never wanted to use INT32_MAX in the first place; I just saw you asserting it in other PRs.

Reply:
@ryanchou-dev what you think

Reply:
@SansPapyrus683 what difference does it make between using INT32_MAX and 1e9? Either way the output is still the same so idk why you think it's necessary to make changes to every single detail. Try submitting the code on AtCoder and see if 1e9 works if that's hard to believe. Also, it's conventional to use 1e9 or wtv in solutions so I don't think it's worth your time or @envyaims time to argue about something that isn't necessary in the first place.

Reply:
so by your logic as long as the output is the same any code is fine?

Reply:
i dont think that's what he's saying??? He's just saying its more conventional to use 1e9 than INT32_MAX so this change isn't necessary and I completely agree.

Reply:
> so by your logic as long as the output is the same any code is fine?

if the code is correct and readable, then it should be fine right?

Reply:
you're lucky i'm a benevolent mod

Reply:
> @ryanchou-dev what you think

errr, as long as it's correctly defined, it doesn't rly matter imo